File: Backend/Terraform/lambda_functions/auth_config.py
import json
import os
import boto3
from botocore.exceptions import ClientError

def lambda_handler(event, context):
    headers = {
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Headers': 'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token',
        'Access-Control-Allow-Methods': 'GET,OPTIONS',
        'Access-Control-Allow-Credentials': 'false'
    }
    
    # Handle preflight OPTIONS request
    if event.get('httpMethod') == 'OPTIONS':
        return {
            'statusCode': 200,
            'headers': headers,
            'body': ''
        }

    try:
        # Get secrets from AWS Secrets Manager
        secretsmanager = boto3.client('secretsmanager')
        secret_arn = os.environ['SECRET_ARN']
        
        response = secretsmanager.get_secret_value(SecretId=secret_arn)
        secrets = json.loads(response['SecretString'])

        # Return public configuration (no secrets)
        config = {
            'auth0': {
                'domain': secrets['auth0_domain'],
                'clientId': secrets['auth0_client_id'],
                'redirectUri': f"https://{os.environ['DOMAIN_NAME']}/",
                'audience': f"https://{secrets['auth0_domain']}/api/v2/",
                'scope': 'openid profile email'
            },
            'google': {
                'clientId': secrets['google_client_id']
            },
            'apiBaseUrl': f"https://api.{os.environ['DOMAIN_NAME']}"
        }

        return {
            'statusCode': 200,
            'headers': headers,
            'body': json.dumps(config)
        }
        
    except ClientError as e:
        print(f"Error fetching auth config: {e}")
        return {
            'statusCode': 500,
            'headers': headers,
            'body': json.dumps({'error': 'Internal server error'})
        }
    except Exception as e:
        print(f"Unexpected error: {e}")
        return {
            'statusCode': 500,
            'headers': headers,
            'body': json.dumps({'error': 'Internal server error'})
        }

File: Backend/Terraform/lambda_packages/temp_auth_config_nodejs/auth_config.js
const { createResponse } = require('./utils/auth');
const AWS = require('aws-sdk');

let cachedAuthConfig = null;

const secretsManager = new AWS.SecretsManager({
  region: process.env.AWS_REGION || 'us-east-1'
});

async function getAuthConfig() {
  if (cachedAuthConfig) {
    return cachedAuthConfig;
  }

  try {
    const secretName = process.env.AUTH_SECRET_NAME;
    if (!secretName) {
      throw new Error('AUTH_SECRET_NAME environment variable not set');
    }

    const result = await secretsManager.getSecretValue({
      SecretId: secretName
    }).promise();

    cachedAuthConfig = JSON.parse(result.SecretString);
    return cachedAuthConfig;
  } catch (error) {
    console.error('Failed to retrieve auth secrets:', error);
    throw error;
  }
}

exports.handler = async (event) => {
  console.log('Auth config request:', JSON.stringify(event, null, 2));

  try {
    // Handle CORS preflight
    if (event.httpMethod === 'OPTIONS') {
      return createResponse(200, { message: 'CORS preflight' });
    }

    // Get auth configuration from Secrets Manager
    const authSecrets = await getAuthConfig();

    // Return Auth0 public configuration
    const config = {
      auth0: {
        domain: authSecrets.auth0_domain,
        clientId: authSecrets.auth0_client_id,
        redirectUri: 'https://vibesheets.com/dashboard',
        audience: authSecrets.auth0_audience,
        scope: 'openid profile email'
      },
      google: {
        clientId: authSecrets.google_client_id
      },
      apiBaseUrl: 'https://api.vibesheets.com'
    };

    return createResponse(200, {
      success: true,
      config
    });

  } catch (error) {
    console.error('Auth config error:', error);
    return createResponse(500, {
      success: false,
      error: 'Internal server error'
    });
  }
};